pkg/scorecard_result.go

```diff
@@ -102,29 +102,47 @@ func (r *ScorecardResult) GetAggregateScore(checkDocs checks.Doc) (float64, erro
 	return score / total, nil
 }
 
+// Define output to console or file.
+func defineOutput(opts *options.Options) (*os.File, error) {
+	if opts.ResultsFile == "" {
+		return os.Stdout, nil
+	}
+	file, err := os.Create(opts.ResultsFile)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create output file: %w", err)
+	}
+	return file, nil
+}
+
 // FormatResults formats scorecard results.
 func FormatResults(
 	opts *options.Options,
 	results *ScorecardResult,
 	doc checks.Doc,
 	policy *spol.ScorecardPolicy,
 ) error {
+	output, outputErr := defineOutput(opts)
+	if outputErr != nil {
+		return outputErr
+	}
+	defer output.Close()
```

Comment:
what happens if `defineOutput` returns `os.Stdout`?
I'm not sure if we use stdout later, but lets avoid closing it.

It may be simpler to just not use the helper function here

Reply:
It logs to console. Ok!